src/actions/files.js

```diff
@@ -36,3 +36,48 @@ export function getFilesForPost(teamId, channelId, postId) {
         ]), getState);
     };
 }
+
+const uploadFileRequests = {};
+export function uploadFile(teamId, channelId, fileFormData, formBoundary, rootId, requestId) {
```

Comment:
Agreed, we can probably find a way to make it work with the redux-store instead of returning a value. That way we can keep using the `await` and `dispatch` like the other actions. Assume we need a ticket for this?